lib/deimos/active_record_consume/batch_record_list.rb

```diff
@@ -23,6 +23,29 @@ def filter!(method)
         self.batch_records.delete_if { |record| !method.call(record.record) }
       end
 
+      # Filter and return removed invalid batch records by the specified method or block
+      # @param method [Proc]
+      # @param block [Proc]
+      # @return [Array<BatchRecord>]
+      def partition!(method=nil, &block)
+        valid, invalid = if method.nil?
```

Comment:
This should never be `nil` since it's defined on the base class, no?

Reply:
Not sure... I think I'd need to see this. It feels like you're diving too deep into Deimos guts to get this working - it feels wrong for calling code to have to call so many internal methods.